src/bootstrap/src/core/config/config.rs

```diff
@@ -2667,11 +2667,14 @@ fn check_incompatible_options_for_ci_rustc(
 ) -> Result<(), String> {
     macro_rules! err {
         ($current:expr, $expected:expr) => {
-            if let Some(current) = $current {
-                if Some(current) != $expected {
+            if let Some(current) = &$current {
+                if Some(current) != $expected.as_ref() {
                     return Err(format!(
-                        "ERROR: Setting `rust.{}` is incompatible with `rust.download-rustc`.",
-                        stringify!($expected).replace("_", "-")
+                        "ERROR: Setting `rust.{}` is incompatible with `rust.download-rustc`. \
+                        Current value: {:?}, Expected value(s): None/{:?}",
```

Comment:
This can print `Expected value(s): None/None` if the expected value is `None`. A bit confusing, but I think it's fine to keep if you don't want to complicate this further.

Reply:
Handled with https://github.com/rust-lang/rust/pull/129052/commits/fb06efd1ef1bf6f6f4f2141119be863d78e0398c